plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/handler/ResultSetHandlerMain.java

```diff
@@ -196,6 +197,7 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
         String actionId = event.getCommand().getId();
         IResultSetPresentation presentation = rsv.getActivePresentation();
         DBPDataSource dataSource = rsv.getDataSource();
+        BasePolicyDataProvider policyProvider = new BasePolicyDataProvider();
```

Comment:
Ne need to create new instance here. This handler processes dozens of different commands.

Reply:
Done. The handler instance moved to class field.